ceph_deploy/hosts/centos/install.py

```diff
@@ -98,6 +98,23 @@ def install(distro, version_kind, version, adjust_repos, **kw):
                 extra_installs=False
             )
 
+        elif version_kind == 'dev_commit':
+            logger.info('skipping install of ceph-release package')
+            logger.info('repo file will be created manually')
+            mirror_install(
+                distro,
+                'http://gitbuilder.ceph.com/ceph-rpm-centos{release}-{machine}-basic/sha1/{version}/'.format(
+                    release=release.split(".", 1)[0],
+                    machine=machine,
+                    version=version),
+                gpg.url(key),
+                adjust_repos=True,
+                extra_installs=False
+            )
+
+        else:
+            raise Exception('unrecognized version_kind %s' % version_kind)
+
```

Comment:
I hate to see all this duplicated code.  :)  Since it looks like the only difference is `ref` vs `sha1` in the URL, can we instead do something like:

```
if version_kind in ['dev', 'dev_commit']:
...

url = 'http://gitbuilder.ceph.com/ceph-rpm-centos{release}-{machine} basic/{type}/{version}/'.format(
    release=release.split(".", 1)[0],
    machine=machine,
    type='ref' if version_kind == 'dev' else 'sha1',
    version=version)

mirror_install(
    distro,
    url,
    gpg.url(key),
    adjust_repos=True,
    extra_installs=False
)

```

Same in Fedora.